src/NHibernate/Dialect/MsSql2000Dialect.cs

```diff
@@ -66,35 +66,412 @@ protected virtual void RegisterDefaultProperties()
 
 		protected virtual void RegisterKeywords()
 		{
-			RegisterKeyword("top");
-			RegisterKeyword("int");
-			RegisterKeyword("integer"); // a commonly-used alias for 'int'
-			RegisterKeyword("tinyint");
-			RegisterKeyword("smallint");
+			RegisterKeyword("absolute");
```

Comment:
Were you concerned by the number of lines it takes, or by the slowness of hundreds of function calls?

If I change a ReSharper array initializer wrap setting, I get something like this:


		private static readonly string[] keywordsToRegister =
		{
			"absolute", "action", "ada", "add", "admin", "after", "aggregate", "alias", "all", "allocate", "alter", "and", "any",
			"are", "array", "as", "asc", "assertion", "at", "authorization", "avg", "backup", "before", "begin", "between",
			"bigint", "binary", "bit", "bit_length", "blob", "boolean", "both", "breadth", "break", "browse", "bulk", "by",
			"call", "cascade", "cascaded", "case", "cast", "catalog", "char", "char_length", "character", "character_length",
			"check", "checkpoint", "class", "clob", "close", "clustered", "coalesce", "collate", "collation", "column", "commit",
			"completion", "compute", "connect", "connection", "constraint", "constraints", "constructor", "contains",
			"containstable", "continue", "convert", "corresponding", "count", "create", "cross", "cube", "current",
			"current_date", "current_path", "current_role", "current_time", "current_timestamp", "current_user", "cursor",
			"cycle", "data", "database", "date", "datetime", "day", "dbcc", "deallocate", "dec", "decimal", "declare", "default",
			"deferrable", "deferred", "delete", "deny", "depth", "deref", "desc", "describe", "descriptor", "destroy",
			"destructor", "deterministic", "diagnostics", "dictionary", "disconnect", "disk", "distinct", "distributed",
			"domain", "double", "drop", "dummy", "dump", "dynamic", "each", "else", "end", "end-exec", "equals", "errlvl",
			"escape", "every", "except", "exception", "exec", "execute", "exists", "exit", "external", "extract", "false",
			"fetch", "file", "fillfactor", "first", "float", "for", "foreign", "fortran", "found", "free", "freetext",
			"freetexttable", "from", "full", "function", "general", "get", "global", "go", "goto", "grant", "group", "grouping",
			"having", "holdlock", "host", "hour", "identity", "identity_insert", "identitycol", "if", "ignore", "image",
			"immediate", "in", "include", "index", "indicator", "initialize", "initially", "inner", "inout", "input",
			"insensitive", "insert", "int", "integer", "intersect", "interval", "into", "is", "isolation", "iterate", "join",
			"key", "kill", "language", "large", "last", "lateral", "leading", "left", "less", "level", "like", "limit", "lineno",
			"load", "local", "localtime", "localtimestamp", "locator", "lower", "map", "match", "max", "min", "minute",
			"modifies", "modify", "module", "money", "month", "names", "national ", "natural", "nchar", "nclob", "new", "next",
			"no", "nocheck", "nonclustered", "none", "not", "ntext", "null", "nullif", "numeric", "nvarchar", "object",
			"octet_length", "of", "off", "offsets", "old", "on", "only", "open", "opendatasource", "openquery", "openrowset",
			"openxml", "operation", "option", "or", "order", "ordinality", "out", "outer", "output", "over", "overlaps", "pad",
			"parameter", "parameters", "partial", "pascal", "path", "percent", "plan", "position", "postfix", "precision",
			"prefix", "preorder", "prepare", "preserve", "primary", "print", "prior", "privileges", "proc", "procedure",
			"public", "raiserror", "read", "reads", "readtext", "real", "reconfigure", "recursive", "ref", "references",
			"referencing", "relative", "replication", "restore", "restrict", "result", "return", "returns", "revoke", "right",
			"role", "rollback", "rollup", "routine", "row", "rowcount", "rowguidcol", "rows", "rule", "save", "savepoint",
			"schema", "scope", "scroll", "search", "second", "section", "select", "sequence", "session", "session_user", "set",
			"sets", "setuser", "shutdown", "size", "smalldatetime", "smallint", "smallmoney", "some", "space", "specific",
			"specifictype", "sql", "sql_variant", "sqlca", "sqlcode", "sqlerror", "sqlexception", "sqlstate", "sqlwarning",
			"start", "state", "statement", "static", "statistics", "structure", "substring", "sum", "system_user", "table",
			"temporary", "terminate", "text", "textsize", "than", "then", "time", "timestamp", "timezone_hour",
			"timezone_minute", "tinyint", "to", "top", "trailing", "tran", "transaction", "translate", "translation", "treat",
			"trigger", "trim", "true", "truncate", "tsequal", "under", "union", "unique", "uniqueidentifier", "unknown",
			"unnest", "update", "updatetext", "upper", "usage", "use", "user", "using", "value", "values", "varbinary",
			"varchar", "variable", "varying", "view", "waitfor", "when", "whenever", "where", "while", "with", "without", "work",
			"write", "writetext", "year", "zone",
		};

		protected virtual void RegisterKeywords()
		{
			RegisterKeywords(keywordsToRegister);
		}

The nice thing about one-per-line was that it could be easily sorted.